test/test_analysis.py

```diff
@@ -13,27 +17,28 @@ class TestAnalysis(DbIsolatedTest):
     async def setUp(self) -> None:
         # don't need to await because it's tagged @run_as_sync
         super().setUp()
-        sl = SampleLayer(self.connection)
+        self.sl = SampleLayer(self.connection)
         self.al = AnalysisLayer(self.connection)
 
-        await sl.insert_sample(
+        self.sample_id = await self.sl.insert_sample(
             'Test01',
             SampleType.BLOOD,
             meta={'meta': 'meta ;)'},
             active=True,
         )
 
+        await self.al.insert_analysis(
+            analysis_type=AnalysisType.CRAM,
+            status=AnalysisStatus.COMPLETED,
+            sample_ids=[self.sample_id],
+            meta={'sequence_type': 'genome', 'size': 1024},
+        )
+
     @run_test_as_sync
     async def test_add_cram(self):
         """
         Test adding an analysis of type CRAM
         """
-        await self.al.insert_analysis(
```

Comment:
Hmm, I think if this behaviour changes and the test fails it will be tricky to narrow down, I'd be more in favour of moving the insert sample into each of these tests (as well as keeping the `insert_analysis` into each test.